ddtrace/contrib/asgi/middleware.py

```diff
@@ -183,4 +189,6 @@ async def wrapped_send(message):
                 del scope["datadog"]["request_span"]
             except KeyError:
                 pass
-            span.finish()
+
+            if span:
```

Comment:
is there any reason why we wouldn't have the span here?

or do you want to check if it is already finished or not?

Reply:
Yes, I was worried about calling `span.finish()` multiple times, since I thought after the first close, it would somehow disappear and cause errors if the span is closed again later. 

I just tested this locally and it seems that the span is still available in the finally block and all the tests for fastapi still pass:

```
(Pdb) span.duration
0.006059125
(Pdb) span.finish()  
(Pdb) span.duration
0.006059125
(Pdb) span.finish()
(Pdb) span.duration
```

Reply:
I removed the `if span` condition since it doesn't seem needed.

Reply:
Marking this as resolved since I removed the `if span` logic.